Purchases/Networking/HTTPClient.swift

```diff
@@ -69,6 +65,71 @@ class HTTPClient {
 
 }
 
+private extension HTTPClient {
+
+    // swiftlint:disable nesting
+    struct Request: CustomStringConvertible {
+
+        typealias Headers = [String: String]
+        typealias RequestBody = [String: Any]
+        typealias Completion = ((_ statusCode: Int, _ response: [String: Any]?, _ error: Error?) -> Void)
+
+        enum Method {
+
+            case get
+            case post(body: RequestBody)
+
+            var httpMethod: String {
+                switch self {
+                case .get: return "GET"
+                case .post: return "POST"
+                }
+            }
+
+        }
+
+        var method: Method
+        var path: String
+        var headers: Headers
+        var completionHandler: Completion?
+        var retried: Bool = false
```

Comment:
I love that if we change this default to `true` a bunch of tests fail 🎉

Reply:
Yessssss